apps/condo/domains/acquiring/schema/CreatePaymentByLinkService.js

```diff
@@ -68,72 +65,31 @@ const CreatePaymentByLinkService = new GQLCustomSchema('CreatePaymentByLinkServi
                 const {
                     PersonalAcc, // organization's bank account
                     BIC,
-                    PayerAddress,
                     PaymPeriod, // mm.yyyy
                     Sum,
-                    PayeeINN,
                     PersAcc, // resident's account within organization
                 } = qrCodeFields
                 const period = formatPeriodFromQRCode(PaymPeriod)
                 const amount = String(Big(Sum).div(100))
 
-                // Stage 1: normalize address
-                const addressServiceClient = createInstance()
-                const normalizedAddress = await addressServiceClient.search(PayerAddress, { extractUnit: true })
-
-                if (!normalizedAddress.addressKey) throw new GQLError(ERRORS.ADDRESS_IS_INVALID, context)
-
-                // Stage 2: find properties
-                const properties = await find('Property', {
-                    organization: { tin: PayeeINN, deletedAt: null },
-                    addressKey: normalizedAddress.addressKey,
-                    deletedAt: null,
-                })
-
-                const organizationsIds = properties.map((item) => item.organization)
-
-                // Stage 3: find organizations with valid contexts and make contexts map
-                const billingContexts = await BillingIntegrationOrganizationContext.getAll(context, {
-                    organization: { id_in: organizationsIds, deletedAt: null },
-                    status: CONTEXT_FINISHED_STATUS,
-                    deletedAt: null,
-                })
-
-                const acquiringContexts = await AcquiringIntegrationContext.getAll(context, {
-                    organization: { id_in: organizationsIds, deletedAt: null },
-                    status: ACQUIRING_CONTEXT_FINISHED_STATUS,
-                    deletedAt: null,
-                })
+                const auxiliaryData = await findAuxiliaryData(qrCodeFields, { address: ERRORS.ADDRESS_IS_INVALID })
+                const { normalizedAddress } = auxiliaryData
 
-                /**
-                 * Map billingContextId to acquiringContextId
-                 * @type {Object<string, string>}
-                 * */
-                const contextsMap = {}
-                for (const billingContext of billingContexts) {
-                    const orgId = get(billingContext, ['organization', 'id'])
-                    for (const acquiringContext of acquiringContexts) {
-                        if (orgId === get(acquiringContext, ['organization', 'id'])) {
-                            contextsMap[billingContext.id] = acquiringContext.id
-                        }
-                    }
-                }
+                const [organizationId] = Object.keys(auxiliaryData.contexts)
+                const { billingContext, acquiringContext } = auxiliaryData.contexts[organizationId]
```

Comment:
Why were all contexts used before, but now only the first one?

Reply:
Will there always be 1 context here?

Reply:
Here will be only one organization. If there are 2+ organization an error will be thrown by `validateQRCode` called above.